graph/patterns/dictionary.md

```diff
@@ -31,41 +31,122 @@ Before using a dictionary type in your API definition, make sure that your scena
+Because dictionary entries are removed by setting the value to `null`, dictionaries don't support null values.
 
 For more information, see the [OData reference](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#dictionary).
 
 ## Examples
 
-### JSON payload example
+### Declaring a string dictionary
+The following example demonstrates defining a dictionary that can contain string values.
+
+```xml
+<ComplexType Name="stringDictionary" BaseType="graph.Dictionary">
+  <Annotation Term="Org.OData.Validation.V1.OpenPropertyTypeConstraint">
+    <Collection>
+      <String>Edm.String</String>
+    </Collection>
+  </Annotation>
+</ComplexType>
+```
+
+### Defining a dictionary property
+The following example shows defining a dictionary property, "userTags", on the item entity type.
+
+```xml
+<EntityType Name="item">
+  ...
+  <Property Name="userTags" Type="self.stringDictionary"/>
+</EntityType>
+```
+
+### Reading a dictionary
+Dictionaries are represented in JSON payloads as a JSON object, where the property names are comprised of the keys and their values are the corresponding key values. 
 
-The following example illustrates the resulting JSON for a property of dictionary type. The parent object has been omitted for brevity.
+The following example shows reading an item with a dictionary property named "userTags":
 
 ```json
+GET item
 {
-  "author": {
-    "domain": "contoso"
-  },
-  "maintainer": {
-    "domain": "fabrikam"
-  },
-  "architect": {
-    "domain": "adventureWorks"
+  ...
+  "userTags":
+  {
+    "anniversary": "2002-05-19",
+    "favoriteMovie": "Princess Bride"
   }
 }
 ```
 
-### HTTP calls examples
+### Setting a dictionary value
+The following example shows setting a dictionary value.  If "hairColor" already exists, it is updated, otherwise it is added.
 
-In this set of examples, we model a **roles** property of dictionary type on the user entity, which is exposed by the users entity set.
+```json
+PATCH item/userTags
+{
+   "hairColor": "purple"
+}
+```
+
+### Deleting a dictionary value
+A dictionary value can be removed by setting the value to null.
+
+```json
+PATCH item/userTags
+{
+   "hairColor": null
+}
+```
 
-#### Get an entry from the dictionary
+### Declaring a complex typed dictionary
+Dictionaries can also contain complex types whose values may be constrained to a particular set of complex types.
+
+The following example defines a complex type **roleSettings**, an **assignedRoleGroupDictionary** that contains **roleSettings**, and an **assignedRoles** property that uses the dictionary..
+
+```xml
+<EntityType Name="principal">
+  ...
+  <Property Name="assignedRoles" Type="self.assignedRoleGroupDictionary">
+</EntityType>
+
+<ComplexType Name="roleSettings">
+  <Property Name ="domain" Type="Edm.String" Nullable="false" />
+</ComplexType>
+
+<ComplexType Name="assignedRoleGroupDictionary" BaseType="graph.Dictionary">
```

Comment:
i believe it should be an open type

Reply:
Dictionary is an open type, so I think this is OK.